m4/sage_spkg_configure.m4

```diff
@@ -185,22 +185,40 @@ AC_DEFUN([SAGE_SPKG_CONFIGURE], [
 
 # SYNOPSIS
 #
-#   SAGE_SPKG_DEPCHECK(PACKAGE-DEPENDENCIES, FURTHER-CHECK)
-#                                 $1              $2
+#   SAGE_SPKG_DEPCHECK(PACKAGE-DEPENDENCIES, FURTHER-CHECK, [FAIL= [sage_spkg_install_]SPKG_NAME=yes])
+#                                 $1              $2                $3
 #
 # DESCRIPTION
 #     *** to be called from SAGE_SPKG_CONFIGURE* ***
 #     check for space-separated list of package dependencies $1 of package SPKG_NAME
 #     do $2 if successful
 #
+#     $3 is for off-label use - checking a reverse dependency. Then the default action of
+#     setting sage_spkg_install_SPKG_NAME=yes may be overwritten, typically to [], i.e. nothing.
+#
 AC_DEFUN([SAGE_SPKG_DEPCHECK], [
     m4_foreach_w([DEP], $1, [
        AC_REQUIRE([SAGE_SPKG_CONFIGURE_]m4_toupper(DEP))])
     AC_MSG_CHECKING([whether any of $1 is installed as or will be installed as SPKG])
     AS_IF([test x = y m4_foreach_w([DEP], $1, [ -o [x$sage_spkg_install_]DEP = xyes])], [
         AC_MSG_RESULT([yes; install SPKG_NAME as well])
-        [sage_spkg_install_]SPKG_NAME=yes], [
+        m4_default($3, [sage_spkg_install_]SPKG_NAME=yes)
+	], [
         AC_MSG_RESULT([no])
         $2
         ])
 ])
+
+
+# SYNOPSIS
+#
+#   SAGE_SPKG_REVDEPCHECK(PACKAGE-REVERSE-DEPENDENCIES)
+#                                 $1
+# DESCRIPTION
+#     *** to be called from SAGE_SPKG_CONFIGURE* ***
+#     check for space-separated list of package reverse dependencies $1 of package SPKG_NAME
+#     If they all come from the system, do not require SPKG_NAME
+#
+AC_DEFUN([SAGE_SPKG_REVDEPCHECK], [
+    SAGE_SPKG_DEPCHECK($1, [[sage_require_]SPKG_NAME=no], [])
```

Comment:
Is this m4-quoted correctly?

Reply:
here yes, but need to pass `[ ]`, not `[]`, an quote `$3` in the call to `m4_define`.